### PR TITLE
Eileen's exercise 6

### PR DESCRIPTION
**Pull request by xa2113**

### CODE REVIEW

File: src/main/java/com/codingnomads/part_06/1_questions [hunk 1]
```diff
@@ -1,22 +1,38 @@
 1) What is Java inheritance?
+Inheritance allows a subclass to add (extend) a superclass when it is declared.
+This means that a subclass inherits all the fields and methods from a superclass.
 
 2) How does inheritance work?
+Place the keyword "extends" after the class name
 
 3) What are some advantages that inheritance provides?
+1. there is no limit to how many levels of inheritance you can create
+2. a subclass can have direct access to all public members of all parent classes
```
Comment (roioteromorales) [inline src/main/java/com/codingnomads/part_06/1_questions]: and protected!

File: src/main/java/com/codingnomads/part_06/1_questions [hunk 2]
```diff
@@ -1,22 +1,38 @@
...
 
 4) What does the "super" keyword do?
+The "super" keyword is used to refer to a variable or method in the parent class.
 
 5) In what order are constructors called when there are multiple subclasses?
```
Comment (roioteromorales) [inline src/main/java/com/codingnomads/part_06/1_questions]: Missing this answer, its from inner to outer: 
A extends B extends C
first constructor would be C, then B then A

File: src/main/java/com/codingnomads/part_06/1_questions [hunk 3]
```diff
@@ -1,22 +1,38 @@
...
 9) What is dynamic method dispatch?
+Dynamic method dispatch is the mechanism by which a call to an overriden method is resolved at run time rather than compile time.
+This is important because this is how Java implements run-time polymorphism.
 
 10) What does the "final" keyword do?
+variables with the final keyword do not change.
 
 11) Are classes in Java are a subclass of what super class?
-
+???
```
Comment (roioteromorales) [inline src/main/java/com/codingnomads/part_06/1_questions]: Object!!! :)

File: src/main/java/com/codingnomads/part_06/Exercises_1To5/Book.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    public Book(){
+    }
+
+    public Book(boolean available, int daysAvailable) {
+        this.available = available;
+        this.daysAvailable = daysAvailable;
+    }
+
+    String setTitle(String title){
```
Comment (roioteromorales) [inline src/main/java/com/codingnomads/part_06/Exercises_1To5/Book.java]: Why this is not public? 💃

**Comment by roioteromorales** [comment]
tenor-172957802

File: src/main/java/com/codingnomads/part_06/1_questions (hunk 1 above)
Comment (xa2113) [reply src/main/java/com/codingnomads/part_06/1_questions]: Ah got it!

File: src/main/java/com/codingnomads/part_06/1_questions (hunk 2 above)
Comment (xa2113) [reply src/main/java/com/codingnomads/part_06/1_questions]: Got it!

File: src/main/java/com/codingnomads/part_06/1_questions (hunk 3 above)
Comment (xa2113) [reply src/main/java/com/codingnomads/part_06/1_questions]: I see... thanks :)

File: src/main/java/com/codingnomads/part_06/Exercises_1To5/Book.java (hunk 4 above)
Comment (xa2113) [reply src/main/java/com/codingnomads/part_06/Exercises_1To5/Book.java]: oh haha right. Changed it!

**Pull request merged**